src/ui_keymap.rs

```diff
@@ -32,17 +37,84 @@ pub fn render_keymap(
   frame.render_widget(block, area);
 
   let items = match state.scope {
-    Scope::Procs => vec![
-      (KeyCode::Char('a'), KeyModifiers::CONTROL, "Toggle focus"),
-      (KeyCode::Char('q'), KeyModifiers::NONE, "Quit"),
-      (KeyCode::Char('j'), KeyModifiers::NONE, "Next"),
-      (KeyCode::Char('k'), KeyModifiers::NONE, "Prev"),
-      (KeyCode::Char('s'), KeyModifiers::NONE, "Start"),
-      (KeyCode::Char('x'), KeyModifiers::NONE, "Stop"),
-      (KeyCode::Char('r'), KeyModifiers::NONE, "Restart"),
-    ],
+    Scope::Procs => {
+      let settings = Settings::default();
+      let prev_default = Key::new(KeyCode::Char('j'), KeyModifiers::NONE);
```

Comment:
We don't need to give special treatment to the default key bindings. The keys are added from three layers:
- defaults
- ~/.config/mproc/mprocs.yaml
- ./mprocs.yaml
All of these layers just override bindings from previous ones. It is possible that the user has overridden some binding with `null` and didn't provide an alternative. Let's just not show anything for that command.

I would suggest to make a vector of events we want to show bindings for, and then iterate through them via `.map_filter` to filter out events without bindings. Something like this:
```rust
vec![
  AppEvent::Quit,
  AppEvent::PrevProc,
  ...
]
...
items.into_iter().filter_map(|event| keymap.resolve_key(event))
```

Reply:
I'm not sure how this would work (from looking at the sample code): The `resolve_key` will return a `Vector` of key bindings - how are we going to decide which key binding has to be shown from that ?

Reply:
I was keeping in mind my previous suggestion to return an Option instead of a Vec.